dhall-json/src/Dhall/JSON.hs

```diff
@@ -279,6 +280,80 @@ instance Show CompileError where
+            \    │     }                                        │ a record value is ok.      \n\
+            \    └──────────────────────────────────────────────┘                            \n\
+            \                                                                                \n\
+            \                                                                                \n\
+            \... is converted to this JSON:                                                  \n\
+            \                                                                                \n\
+            \                                                                                \n\
+            \    ┌─────────────────────────────────┐                                         \n\
+            \    │ { \"name\": \"Record\", \"x\": true } │                                   \n\
+            \    └─────────────────────────────────┘                                         \n\
+            \                                                                                \n\
+            \                                                                                \n\
+            \This isn't valid:                                                               \n\
+            \                                                                                \n\
+            \                                                                                \n\
+            \    ┌──────────────────────────────────────────┐                                \n\
+            \    │ let Example = < Foo : Bool >             │                                \n\
+            \    │                                          │                                \n\
+            \    │ let Nesting = < Inline | Nested : Text > │                                \n\
+            \    │                                          │                                \n\
+            \    │ in  { field = \"name\"                     │                              \n\
+            \    │     , nesting = Nesting.Inline           │                                \n\
+            \    │     , contents = Example.Foo True        │ ❰True❱ is not a record         \n\
+            \    │     }                                    │                                \n\
+            \    └──────────────────────────────────────────┘                                \n\
+            \                                                                                \n\
+            \                                                                                \n\
+            \The following Dhall expression could not be translated to JSON:                 \n\
+            \                                                                                \n\
+            \↳ " <> Core.pretty record <> "                                                  \n\
```

Comment:
The pretty-printing here (and probably in the other errors too) needs improvement. Currently it looks like this:

```
The following Dhall expression could not be translated to JSON:                 
                                                                                
↳ { contents = < Left : { foo : Natural } | Middle : Natural | Right : { bar : Bool } >.Middle 0, field = "name", nesting = < Inline | Nested : Text >.Inline }                                                  
                                                                                
... because                                                                     
                                                                                
↳ 0                                     
                                                                                
... is not a record.
```

Reply:
You can use `Dhall.Util.insert`, which is the same thing type-checking error messages use:

https://github.com/dhall-lang/dhall-haskell/blob/4c1736b1546ff24d7a78734d09866f77ee7944eb/dhall/src/Dhall/Util.hs#L76-L79

Reply:
Thanks, I'm using `Dhall.Util.insert` now. I had to expose `Dhall.Util` from `dhall` though. I hope that's ok.

Reply:
Yeah, that's fine